### ISSUE TITLE
Could not check if ACCESS_NETWORK_STATE permission is available warning in 2.17.1

### ISSUE THREAD

**Issue by RecargoJereld** [aws-amplify/aws-sdk-android#2063]
**Describe the bug**
`Could not check if ACCESS_NETWORK_STATE permission is available` warning is shown in logcat for 2.17.1. It looks like AWSMobileClient is looking for the android.support.v4.content.ContextCompat class but it has been replaced with AndroidX in 2.17.
 
**To Reproduce**
Updated aws-sdk-android to version 2.17.1 and app calls awsMobileClient.currentUserState()

```
com.my.app W/AWSMobileClient: Could not check if ACCESS_NETWORK_STATE permission is available.
    java.lang.ClassNotFoundException: android.support.v4.content.ContextCompat
        at java.lang.Class.classForName(Native Method)
        at java.lang.Class.forName(Class.java:454)
        at java.lang.Class.forName(Class.java:379)
        at com.amazonaws.mobile.client.AWSMobileClient.isNetworkAvailable(AWSMobileClient.java:771)
        at com.amazonaws.mobile.client.AWSMobileClient.getUserStateDetails(AWSMobileClient.java:983)
        at com.amazonaws.mobile.client.AWSMobileClient$5.run(AWSMobileClient.java:845)
        at com.amazonaws.mobile.client.AWSMobileClient$5.run(AWSMobileClient.java:842)
        at com.amazonaws.mobile.client.internal.ReturningRunnable.await(ReturningRunnable.java:31)
        at com.amazonaws.mobile.client.AWSMobileClient.currentUserState(AWSMobileClient.java:824)
```

**Comment by hexdecimal16** [comment]
I am also experiencing a similar issue.

**Comment by hexdecimal16** [comment]
I think this error is because of some other underlying error.

**Comment by WaddySam** [comment]
Hey guys @hexdecimal16 @RecargoJereld , 
Did you find anything for this?

**Comment by hexdecimal16** [comment]
> Hey guys @hexdecimal16 @RecargoJereld ,
> Did you find anything for this?

Unfortunately no, I switched to firebase, for now, it works better for android I guess. But most probably this is because of some other error. Try making a new project and set up properly.

**Comment by WaddySam** [comment]
> > Hey guys @hexdecimal16 @RecargoJereld ,
> > Did you find anything for this?
> 
> Unfortunately no, I switched to firebase, for now, it works better for android I guess. But most probably this is because of some other error. Try making a new project and set up properly.

Thanks a lot. That's sad, I've been trying to complete a personal for a year and I always run into those problems. I might, as well, switch to firebase.

**Comment by rjuliano** [comment]
What Android version are you using? For Android 6.0 (API version 23) and higher, you have to request permission at runtime in addition to adding it to the manifest.

https://developer.android.com/guide/topics/permissions/overview#runtime_requests_android_60_and_higher

**Comment by RecargoJereld** [comment]
@hexdecimal16 I'm using v2.16.13 in the mean time.

@rjuliano I don't think it's a permissions issue. The exception thrown is a ClassNotFoundException. A SecurityException is thrown when a call requires a permission and the permission has not been granted. ACCESS_NETWORK_STATE is a normal level permission and doesn't require runtime approval.

**Comment by rjuliano** [comment]
Out of curiosity, have you tried adding a dependency on the android support library for whatever android version you're using?

**Comment by saltonmassally** [comment]
AWSMobileClient is looking for the `android.support.v4.content.ContextCompat` class but it has been replaced with AndroidX. A possible fix is also to try to look for `androidx.core.content.ContextCompat` if the former returns no class found. I suspect the API for both are similar if not the same (disclaimer - highly uninformed take); if that is so, it should be an easy fix

**Comment by orestesgaolin** [comment]
Does [2.18.0 release](https://github.com/aws-amplify/aws-sdk-android/releases/tag/release_v2.18.0) contain a fix for this particular issue?

**Comment by saltonmassally** [comment]
Does not, offending line is  [here](https://github.com/aws-amplify/aws-sdk-android/blob/main/aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java#L771)

**Comment by AlexeyKorshun** [comment]
You can use android.enableJetifier=true in gradle.properties before an issue will fixed

**Comment by AntoEko** [comment]
> You can use android.enableJetifier=true in gradle.properties before an issue will fixed

It doesn't solve the issue.

**Comment by jamesonwilliams** [comment]
> What is interesting is that the call to `ContextCompat.checkSelfPermission` uses the imported `androidx.core.content.ContextCompat` so I fail to understand what `Class.forName("android.support.v4.content.ContextCompat")` does in that block.

Thank you @saltonmassally for identifying the issue; I agree with you that this line appears to have no purpose. Here's a PR to remove it. https://github.com/aws-amplify/aws-sdk-android/pull/2091/files

**Comment by yoonhok524** [comment]
I am using the 2.18.0.
But I still can find the warning log.

```
2020-09-09 12:43:19.493 2686-2870/co.spoonme W/AWSMobileClient: Could not check if ACCESS_NETWORK_STATE permission is available.
    java.lang.ClassNotFoundException: android.support.v4.content.ContextCompat
        at java.lang.Class.classForName(Native Method)
        at java.lang.Class.forName(Class.java:454)
        at java.lang.Class.forName(Class.java:379)
        at com.amazonaws.mobile.client.AWSMobileClient.isNetworkAvailable(AWSMobileClient.java:771)
        at com.amazonaws.mobile.client.AWSMobileClient.getUserStateDetails(AWSMobileClient.java:983)
        at com.amazonaws.mobile.client.AWSMobileClient.waitForSignIn(AWSMobileClient.java:923)
        at com.amazonaws.mobile.client.AWSMobileClient.getCredentials(AWSMobileClient.java:364)
        at com.amazonaws.services.pinpoint.AmazonPinpointClient.invoke(AmazonPinpointClient.java:4068)
        at com.amazonaws.services.pinpoint.AmazonPinpointClient.putEvents(AmazonPinpointClient.java:3118)
        at com.amazonaws.mobileconnectors.pinpoint.internal.event.EventRecorder.submitEventsAndEndpoint(EventRecorder.java:384)
        at com.amazonaws.mobileconnectors.pinpoint.internal.event.EventRecorder.submitEventsAndEndpoint(EventRecorder.java:363)
        at com.amazonaws.mobileconnectors.pinpoint.internal.event.EventRecorder.processEvents(EventRecorder.java:334)
        at com.amazonaws.mobileconnectors.pinpoint.internal.event.EventRecorder$1.run(EventRecorder.java:247)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.v4.content.ContextCompat" on path: DexPathList[[zip file "/data/app/co.spoonme-K4UMYQeugasD7RTbSo9AEQ==/base.apk"],nativeLibraryDirectories=[/data/app/co.spoonme-K4UMYQeugasD7RTbSo9AEQ==/lib/arm64, /data/app/co.spoonme-K4UMYQeugasD7RTbSo9AEQ==/base.apk!/lib/arm64-v8a, /system/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:196)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at java.lang.Class.classForName(Native Method) 
        at java.lang.Class.forName(Class.java:454) 
        at java.lang.Class.forName(Class.java:379) 
        at com.amazonaws.mobile.client.AWSMobileClient.isNetworkAvailable(AWSMobileClient.java:771) 
        at com.amazonaws.mobile.client.AWSMobileClient.getUserStateDetails(AWSMobileClient.java:983) 
        at com.amazonaws.mobile.client.AWSMobileClient.waitForSignIn(AWSMobileClient.java:923) 
        at com.amazonaws.mobile.client.AWSMobileClient.getCredentials(AWSMobileClient.java:364) 
        at com.amazonaws.services.pinpoint.AmazonPinpointClient.invoke(AmazonPinpointClient.java:4068) 
        at com.amazonaws.services.pinpoint.AmazonPinpointClient.putEvents(AmazonPinpointClient.java:3118) 
        at com.amazonaws.mobileconnectors.pinpoint.internal.event.EventRecorder.submitEventsAndEndpoint(EventRecorder.java:384) 
        at com.amazonaws.mobileconnectors.pinpoint.internal.event.EventRecorder.submitEventsAndEndpoint(EventRecorder.java:363) 
        at com.amazonaws.mobileconnectors.pinpoint.internal.event.EventRecorder.processEvents(EventRecorder.java:334) 
        at com.amazonaws.mobileconnectors.pinpoint.internal.event.EventRecorder$1.run(EventRecorder.java:247) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
```

What should I do?

**Comment by jamesonwilliams** [comment]
@richardmcclellan Are we doing a release this week? This issue has been fixed for 15 days; can we do a 2.18.1?

**Comment by richardmcclellan** [comment]
Yes, I am planning to do a release, hopefully tomorrow, if not Friday, so the fix for this should be in it!

**Comment by richardmcclellan** [comment]
2.19.0 has been released: https://github.com/aws-amplify/aws-sdk-android/releases/tag/release_v2.19.0

**Comment by azisoli** [comment]
@richardmcclellan  we have the same problem in aws amplify for android when i'm using graphql . would you please fix that as well i'm getting same error

**Comment by azisoli** [comment]
I think you need to use tis new version in your amplify-android project and make a release as we have same problem there .

**Comment by jamesonwilliams** [comment]
@azisoli The next Amplify Android release will reference the SDK version 2.19.0.

**Comment by sebsto** [comment]
@jamesonwilliams any ETA ? or are you waiting for other changes to be released at the same time as that one?

**Comment by azisoli** [comment]
@jamesonwilliams and any estimation when this release will happen?

**Comment by raphkim** [comment]
Hi @sebsto @azisoli ,

We are currently waiting on other important bug fixes and improvements to be merged into the mainline before releasing Amplify 1.4.0. We cannot give an exact timeline, but the release will be coming very soon!

**Comment by azisoli** [comment]
@raphkim  i'm using api(graphql) and  i'm calling Amplify.API.mutate(ModelMutation.update(profile) )
                and it return response with prevoius data and when i check dynamodb there is no changes in database  on cloud and the only error that i receive in logs  is this one that we mention here . so could this be the problem? or should be something else?

**Comment by raphkim** [comment]
@azisoli Please feel free to re-open aws-amplify/amplify-android#827 if the issue you are experiencing is not resolved after the release.

**Comment by sebsto** [comment]
@raphkim @jamesonwilliams  Amplify Android 1.3.2 fixes this for me - thanks

**Comment by azisoli** [comment]
@sebsto  i've tried new version but again update is not working for me Amplify.API.mutate(   ModelMutation.update(profile))
it returns result with previous data and i see these in result:
        ((GraphQLResponse)response).data=Expecting an element; looking at ERROR_ELEMENT '(1,27) in /fragment.kt

((java.util.ArrayList)((PaginatedResult)((GraphQLResponse)response).data).items).get(0) =Expecting an element; looking at ERROR_ELEMENT '(1,67) in /fragment.kt

**Comment by richardmcclellan** [comment]
@azisoli that looks like a different issue.  Could you open a separate issue in the amplify-android repo? https://github.com/aws-amplify/amplify-android/issues/new
